deltacat/utils/daft.py

```diff
@@ -0,0 +1,59 @@
+import logging
+from typing import Optional, List
+
+from daft.table import Table
+from daft.logical.schema import Schema
+from daft import TimeUnit
+from daft.io import IOConfig, S3Config
+import pyarrow as pa
+
+from deltacat import logs
+from deltacat.utils.common import ReadKwargsProvider
+
+from deltacat.types.media import ContentType
+
+logger = logs.configure_deltacat_logger(logging.getLogger(__name__))
+
+
+def daft_s3_file_to_table(
+    s3_url: str,
+    content_type: str,
+    content_encoding: str,
+    include_columns: Optional[List[str]] = None,
+    pa_read_func_kwargs_provider: Optional[ReadKwargsProvider] = None,
+    **s3_client_kwargs,
+):
+    assert (
+        content_encoding == ContentType.PARQUET.value
+    ), "daft native reader currently only supports parquet"
+    kwargs = {}
+    if pa_read_func_kwargs_provider is not None:
+        kwargs = pa_read_func_kwargs_provider(content_type, kwargs)
+
+    coerce_int96_timestamp_unit = TimeUnit.from_str(
+        kwargs.get("coerce_int96_timestamp_unit", "ms")
+    )
+
+    table = Table.read_parquet(
+        path=s3_url,
+        columns=include_columns,
```

Comment:
The default behavior for daft is that it will read all of the columns from the parquet file. It looks like the current behavior for pyarrow also just uses include columns: https://github.com/ray-project/deltacat/blob/main/deltacat/utils/pyarrow.py#L231

Reply:
We recently changed the behavior, if column_names is also None, we can read all columns.

Reply:
fixed!